build.cake

```diff
@@ -747,6 +767,15 @@ Task("Publish-NuGet")
     Information("Publish-NuGet Task failed, but continuing with next Task...");
 });
 
+Task("Publish-Coverage")
+    .IsDependentOn("Run-Unit-Tests")
+    .WithCriteria(() => isRunningOnAppVeyor)
```

Comment:
I would add some additional criteria so we do not push coverage for PR and only from main repo and branch:
```csharp
    .WithCriteria(() => !isLocalBuild)
    .WithCriteria(() => !isPullRequest)
    .WithCriteria(() => isMainRepo)
    .WithCriteria(() => isMasterBranch)
```